src/main/java/de/tum/cit/aet/artemis/programming/service/ProgrammingExerciseBuildConfigService.java

```diff
@@ -0,0 +1,161 @@
+     *
+     * @param list the list of key-value pairs
+     * @return a {@link DockerRunConfig} object initialized with the parsed flags, or {@code null} if an error occurs
+     */
+    @Nullable
+    DockerRunConfig getDockerRunConfigFromParsedList(List<List<String>> list) {
+        final int keyIndex = 0;
+        final int valueIndex = 1;
+        try {
+            boolean networkDisabled = false;
+            List<String> env = null;
+            for (List<String> entry : list) {
+                if (entry.size() != 2 || StringUtils.isBlank(entry.get(valueIndex)) || StringUtils.isBlank(entry.get(keyIndex))
+                        || !DockerRunConfig.AllowedDockerFlags.isAllowed(entry.get(keyIndex))) {
+                    log.warn("Invalid Docker flag entry: {}. Skipping.", entry);
+                    continue;
+                }
+                switch (entry.get(keyIndex)) {
+                    case "network":
+                        networkDisabled = entry.get(valueIndex).equalsIgnoreCase("none");
+                        break;
+                    case "env":
+                        env = parseEnvVariableString(entry.get(valueIndex));
+                        break;
+                    default:
+                        log.error("Invalid Docker flag entry: {}. Skipping.", entry);
+                        break;
+                }
+
+            }
+            return new DockerRunConfig(networkDisabled, env);
+        }
+        catch (Exception e) {
+            log.error("Failed to parse DockerRunConfig from JSON string: {}. Using default settings.", list, e);
+        }
+
+        return null;
+    }
+
+    /**
+     * Parses the JSON string representing Docker flags into a list of key-value pairs.
+     *
+     * @return a list of key-value pairs, or {@code null} if an error occurs
+     */
+    @Nullable
+    List<List<String>> parseDockerFlags(ProgrammingExerciseBuildConfig buildConfig) {
+        if (StringUtils.isBlank(buildConfig.getDockerFlags())) {
+            return null;
+        }
+
+        try {
+            ObjectMapper objectMapper = new ObjectMapper();
+
+            return objectMapper.readValue(buildConfig.getDockerFlags(), new TypeReference<>() {
+            });
+        }
+        catch (Exception e) {
+            log.error("Failed to parse DockerRunConfig from JSON string: {}. Using default settings.", buildConfig.getDockerFlags(), e);
+        }
+
+        return null;
+    }
+
+    private List<String> parseEnvVariableString(String envVariableString) {
+        Pattern pattern = Pattern.compile(
+                // match key-value pairs, where the key can be a single word or a string in single or double quotes
+                // key-value pairs are separated by commas
+                "(?:'([^']+)'|\"([^\"]+)\"|(\\w+))=(?:'([^']*)'|\"([^\"]*)\"|([^,]+))");
```

Comment:
(non-blocker, okay for next PR extending Docker run configs) This pattern needs to be recompiled for every environment variable each time it is parsed after loading it from the database. I think it would be better to declare is as class-wide constant attribute instead to avoid this.